src/write/line.rs

```diff
@@ -632,6 +647,24 @@ impl LineProgram {
                 if self.file_has_md5 {
                     w.write(&info.md5)?;
                 }
+                match (self.file_has_embedded_source, info.embedded_source.as_ref()) {
+                    (true, Some(embedded_source @ LineString::LineStringRef(..))) => {
+                        embedded_source.write(
+                            w,
+                            constants::DW_FORM_line_strp,
+                            self.encoding,
+                            debug_line_str_offsets,
+                            debug_str_offsets,
+                        )?;
+                    }
+                    (true, Some(ref val)) => {
+                        panic!("Can only write LineString::LineStringRef for embedded sources, got {:?}", val);
```

Comment:
If `LineStringRef` is the only option, then no point using `LineString`? Just use the reference directly.

Reply:
Was considering that. I wasn't sure if there is already a pattern in gimli for this.

Reply:
Looking at the DWARF issue, it doesn't restrict the string form, so probably shouldn't here either.

Reply:
The LLVM extension however is restricted: 

![image](https://user-images.githubusercontent.com/7396/59669431-7be41300-91ba-11e9-856d-ad17ce8eddbb.png)

Which I assume makes sense because with the form descriptor system you need to match them across all files.

Reply:
LLVM permits `DW_FORM_string` too: [source](https://github.com/llvm-mirror/llvm/blob/d65677457c49ce6b6afd3a287793b59b56a9b20e/lib/MC/MCDwarf.cpp#L429). I don't see a good reason to restrict the form in gimli when supporting all can be done by simply calling `LineString::write`.

Reply:
I can do that, but i need to know which type i'm going to emit for the form descriptor before I see the file info. So which one do I pick? The string type of the first file info's source and then require them all to match?

Reply:
Yes. That's what the `file_form` handling above does.